### PR TITLE
Classic Block: Fix TinyMCE custom buttons width

### LINKED ISSUE #4360: Transition for TinyMCE styleselect with custom styles
## Issue Overview
With the classic editor, we can add [custom styles into a dropdown](https://codex.wordpress.org/TinyMCE_Custom_Styles) to facilitate the use of custom styles for the user.

capture d ecran 2018-01-08 a 23 04 41

What will happen to these custom styles?
Will they be automatically converted in some way to blocks or will it be a manual work?

## Current Behavior
In the _Classic Text_ block, there's a style conflict with the _Formats_ dropdown.

capture d ecran 2018-01-08 a 23 05 43

When we open the dropdown, there are no styles available.

capture d ecran 2018-01-08 a 23 07 46

### PR DESCRIPTION
**Pull request by youknowriad**
closes #4360

Some TinyMCE buttons like the "formats" button is wider than a regular icon. This PR uses min-width for the TinyMCE buttons to fix these buttons.

**Testing instructions**

 * Enable the formats button like so:

```php
add_filter( 'mce_buttons_2', 'fb_mce_editor_buttons' );
function fb_mce_editor_buttons( $buttons ) {
	array_unshift( $buttons, 'styleselect' );
	return $buttons;
}
```

 * Add a classic block

Verfiy that you can apply the custom formats.

### CODE REVIEW

File: lib/client-assets.php [hunk 1]
```diff
@@ -696,63 +696,70 @@ function gutenberg_editor_scripts_and_styles( $hook ) {
...
+			'redo',
+			'wp_help',
+		), 'editor' ) ),
+		'toolbar3'         => implode( ',', apply_filters( 'mce_buttons_3', array(), 'editor' ) ),
+		'toolbar4'         => implode( ',', apply_filters( 'mce_buttons_4', array(), 'editor' ) ),
+		'external_plugins' => apply_filters( 'mce_external_plugins', array() ),
+	), 'editor' );
+	if ( isset( $tinymce_settings['style_formats'] ) && is_string( $tinymce_settings['style_formats'] ) ) {
+		// Decode the options as we used to recommende json_encoding the TinyMCE settings.
+		$tinymce_settings['style_formats'] = json_decode( $tinymce_settings['style_formats'] );
```
Comment (iseulde) [inline lib/client-assets.php]: Do you have an example and documentation that uses this?

Comment (iseulde) [reply lib/client-assets.php]: Wondering because I can't find the code in core that decodes this part specifically.

Comment (iseulde) [reply lib/client-assets.php]: I mean, there is some weird parsing going on there, so it might be that this affects more than just this setting.

Comment (youknowriad) [reply lib/client-assets.php]: Sure, this is based on the example showcased in the codex https://codex.wordpress.org/TinyMCE_Custom_Styles

**Review by ellatrix — APPROVED** [review]

**Pull request merged**